### DISCUSSION TITLE
How to expose API methods to metadata page?

### DISCUSSION THREAD

**Discussion by danreg** [CoreWCF/CoreWCF#1365] [Q&A]
I'm trying to migrate from an old WCF project to CoreWCF.
I have implemented the first method now, but the method is not shown on the metadata page.
In my original asmx file I have [WebMethod(Description="Bla")] Attributes. Is that anyhow available in CoreWCF?

**Comment by mconnew** [comment]
WebMethod is used with the old .NET 2.0 WebServices library. Your fastest path to moving to CoreWCF is to expose a wsdl on your existing service, then use either the .NET Framework svcutil tool or the .NET dotnet-svcutil tool to create a client which will include the binding and service contract interface along with and data contract data types if needed. You can copy the service contract interface and data contract types into a CoreWCF project and implement the interface. The shape of methods might change slightly, for example if you are passing any data via SOAP headers, that is done a different way (using MessageContract instead) but you should be able to match things up without difficulty. Then copy your service implementation across to the new methods.

If you have difficulty with a specific mismatch in how things are done, feel free to ask and mention me in your comment and I will try to help.

**Reply by danreg** [reply]
@mconnew thx for your reply, I will try this approach.
Meanwhile I think I need to clarify what I meant to say. If I call the old service in the browser then I get a page containing all available methods including descriptions.

Like this
<img width="620" alt="image" src="https://github.com/CoreWCF/CoreWCF/assets/16964633/3996392d-8e6b-4d1a-8b7d-1355bd1d198a">
I'm wondering if this is also possible with corewcf and some attributes on the interface class?
